### PR TITLE
Refactors python operator entrypoints

### PR DESCRIPTION
**Pull request by saurav-c**
## Describe your changes and why you are making these changes
This PR refactors the entrypoints to all of our Python operator implementations. The only way to invoke the Python operators was to execute the relevant Python module and pass in the spec as a command line argument. This will not work for Airflow, as the entrypoint needs to be a Python function, not a bash script. These changes will enable Airflow support (for which there will be several follow on PRs.)

The high level changes are as follows:
- `main.py` has for the most part been renamed to `execute.py`
- `main.py` contains only the code for parsing the spec from the command line argument. It then calls the `run` function in `execute.py` which does the core logic of the operator and is responsible for capturing user errors and reporting the execution state to storage as before.

## Related issue number (if any)
ENG 1392

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/python/aqueduct_executor/operators/connectors/tabular/spec.py [hunk 1]
```diff
@@ -44,10 +44,12 @@ def unwrap_connector_config(cls, connector_config, values):  # type: ignore
     if "connector_name" not in values:
         raise ValueError("Unknown connector name.")
 
-    values["connector_name"]
-
     if not isinstance(connector_config, dict):
         raise ValueError("connector_config is not a dictionary.")
+    
+    if "conf" not in connector_config:
```
Comment (saurav-c) [inline src/python/aqueduct_executor/operators/connectors/tabular/spec.py:50]: This is necessary for Airflow parsing of the spec

**Comment by likawind** [comment]
To make sure I understand, the `run()` will be what's shared between both airflow and our own executor? And in this case, we will pass both the `run()` to airflow, and job spec will be generated as airflow orchestrate the DAG?

**Comment by saurav-c** [comment]
@likawind Yes, that's correct

**Comment by saurav-c** [comment]
@ reviewers the integration tests are failing because the `start-function-executor.sh` script was modified, and the integration tests are still using the old script.

File: src/python/aqueduct_executor/operators/connectors/tabular/spec.py (hunk 1 above)
Comment (kenxu95) [reply src/python/aqueduct_executor/operators/connectors/tabular/spec.py:52]: Can you add a comment explaining why this is custom logic for Airflow?

**Review by kenxu95 — APPROVED** [review]
Looks fine to me 👍

**Review by cw75 — APPROVED** [review]

**Pull request merged**